src/lib/operators/join_sort_merge.cpp

```diff
@@ -930,10 +930,18 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
     // the hash join, we do not (for now) merge small partitions to keep the sorted chunk guarantees, which could be
     // exploited by subsequent operators.
     constexpr auto ALLOW_PARTITION_MERGE = false;
-    auto output_chunks =
-        write_output_chunks(_output_pos_lists_left, _output_pos_lists_right, _left_input_table, _right_input_table,
-                            create_left_side_pos_lists_by_segment, create_right_side_pos_lists_by_segment,
-                            OutputColumnOrder::LeftFirstRightSecond, ALLOW_PARTITION_MERGE);
+    const auto expected_partition_count = std::ranges::count_if(_output_pos_lists_left, [&](auto& list) {
+      return !list.empty();
+    });
+    auto output_segments = std::vector<Segments>(expected_partition_count);
+    output_segments.resize(write_output_segments<ALLOW_PARTITION_MERGE>(
```

Comment:
Reading-wise, it would be helpful to have a telling name for the return value and use this to resize. It's a line more, but helps following.
And can you please Assert (it's cheap enough) that both values (left and right) are equal?

Reply:
Same above in the hash join.